packages/react-router-dom/dom.ts

```diff
@@ -175,8 +179,16 @@ export function getFormSubmissionInfo(
       options as any
     ).submissionTrigger;
 
+    if (options.action) {
+      action = options.action;
+    } else {
+      // When grabbing the action from the element, it will have had the basename
+      // prefixed to ensure non-JS scenarios work, so strip it since we'll
+      // re-prefix in the router
+      let attr = target.getAttribute("action");
+      action = attr ? stripBasename(attr, basename) : null;
+    }
```

Comment:
A few things going on here:
* We now fallback to `action:null` if nothing is available, which lets the router handle differing behavior for a missing action on submission navigations via `<Form>`
* We want our `<form action>`/`<button formaction>` values to contain the `basename` if it exists for non-JS, but since the router will handle prepending the `basename` now we strip it back off on our resolved `action`
* If a user if providing an action in options it's not expected to have a `basename` (same as `<Form action>`)